src/components/common/Component.ts

```diff
@@ -0,0 +1,285 @@
+        }
+      });
+    }
+
+    if (this._parentElement) {
+      this._parentElement.innerHTML = '';
+      if (this.element) {
+        this._parentElement.appendChild(this.element);
+      }
+    }
+  }
+
+  /** Получить строку для рендера компонента */
+  render(): string {
+    const templator = new Templator(this.template);
+    return templator.compile(this.props);
+  }
+
+  /** Функция при включении событий-обработчиков */
+  enableEvents(): void {
+    this._eventBus.emit(ComponentEvents.FLOW_EE);
+  }
+
+  /** Триггер: были включены события-обработчики */
+  private _enableEvents(): void {
+    if (this.element) {
+      ComponentEventTypes.forEach((ev) => {
+        const elements = this.element.querySelectorAll(`[c-on-${ev}]`);
+        if (elements.length > 0) {
+          elements.forEach((el) => {
+            const element = el as HTMLElement;
+            const attr = element.getAttribute(`c-on-${ev}`);
+            if (attr.trim()) {
+              const event = get(this.props, attr.trim());
+              if (event) {
+                if (!this.events.has(element)) {
+                  this.events.set(element, []);
+                }
+                this.events.get(element).push({
+                  event: ev,
+                  func: event as EventListener,
+                });
+              }
+            }
+          });
+        }
+      });
+    }
+
+    if (Array.from(this.events.keys()).length > 0) {
+      const elements = Array.from(this.events.keys());
+      elements.forEach((el) => {
+        if (this.events.has(el)) {
+          const events = this.events.get(el);
+          events.forEach((ev) => {
+            el.addEventListener(ev.event, ev.func.bind(this));
+          });
+        }
+      });
+    }
+  }
+
+  /** Функция при выключении обработчиков */
+  disableEvents(): void {
+    this._eventBus.emit(ComponentEvents.FLOW_ED);
+  }
+
+  /** Триггер: события-обработчики были выключены */
+  private _disableEvents(): void {
+    if (Object.keys(this.childs).length > 0) {
+      Object.keys(this.childs).forEach((guid) => {
+        this.childs[guid].component.disableEvents();
+      });
+    }
+
+    if (this.events.keys.length > 0) {
+      const elements = Array.from(this.events.keys());
+      elements.forEach((el) => {
+        if (this.events.has(el)) {
+          const events = this.events.get(el);
+          events.forEach((ev) => {
+            el.removeEventListener(ev.event, ev.func);
+          });
+        }
+        this.events.delete(el);
+      });
+    }
+  }
+
+  /** Получить текущий элемент */
+  getContent(): HTMLElement {
+    return this.element;
+  }
+
+  /**
+     * Сделать из объекта пропсы на проксях
+     * @param props Объект
+     */
+  private _makePropsProxy(props): void {
+    const self = this;
+
+    this.props = new Proxy(props, {
+      get(obj, prop) {
+        return obj[prop];
+      },
+      set(obj, prop, val) {
```

Comment:
Можно использовать стрелочные функции, тогда не придется замыкать `this`
```suggestion
      set: (obj, prop, val) => {
```